File: server/src/RentnRoll.Domain/Entities/Rentals/Rental.cs
using RentnRoll.Domain.Common;
using RentnRoll.Domain.Entities.BusinessGames;
using RentnRoll.Domain.Entities.Rentals.Enums;
using RentnRoll.Domain.ValueObjects;

namespace RentnRoll.Domain.Entities.Rentals;

public class Rental : Entity
{
    public string UserId { get; set; } = null!;
    public RentalStatus Status { get; set; } = RentalStatus.Expectation;

    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int TotalPrice { get; set; }

    public LockerRental? LockerRental { get; set; }
    public StoreRental? StoreRental { get; set; }
}

File: client/src/features/rentals/types/Rental.ts
export type Rental = {
  id: string;
  status: string;
  startDate: Date;
  endDate: Date;
  totalPrice: number;
  address: string;
  gameName: string;
  gameThumbnail: string;
  locationName: string;
  iotDeviceId: string | null;
};
